tests/regression/39-signed-overflows/06-shiftleft.c

```diff
@@ -0,0 +1,26 @@
+// PARAM: --enable ana.int.interval
+#include <goblint.h>
+int main()
+{
+    int r;
+    int zero_or_one = 0;
+    int top;
+    char c;
+    r = c << 1U; //NOWARN
+
+    r = c << 128U; //WARN
+    r = r << 1U; //WARN
+    r = 8 << -2; //WARN
+
+    if(top) { zero_or_one = 1; }
+
+    r = 8 << zero_or_one;
+
+    __goblint_check(r >= 8);
+    __goblint_check(r <= 16);
+
+    int regval;
+    unsigned long bla = (unsigned long )((1 << ((int )regval >> 6)) << 20);  //WARN
```

Comment:
Doesn't the subexpression `(1 << ((int )regval >> 6)` already overflow potentially? If so, one may add a line for that.

Reply:
Due to #1258, this might only be warning due to the cast right now?

Reply:
The subexpression actually already potentially overflows and Goblint detects it; I extended the test case accordingly.